casper/src/main/scala/coop/rchain/casper/api/BlockApiImpl.scala

```diff
@@ -175,6 +183,81 @@ class BlockApiImpl[F[_]: Concurrent: RuntimeManager: BlockDagStorage: BlockStore
       )
   }
 
+  override def deployStatus(deployId: DeployId): F[ApiErr[DeployExecStatus]] = {
```

Comment:
Very elegant implementation! It is easy to read and understand but difficult to write for me :slightly_smiling_face:
It is interesting to use `collect` and `as` functions. I always use `map` but your solution with this functions more beautiful :+1: